### PR TITLE
VerletClusterLists Coloring SoA

### LINKED ISSUE #311: SoA for VerletClusterLists Coloring
**Is your feature request related to a problem? Please describe.**
The coloring traversal for `VerletClusterLists` (#295) does not support `SoA` yet.

**Describe the solution you'd like**
It should support `SoA`.

#298 First implementation.

### PR DESCRIPTION
**Pull request by HobbyProgrammer**
# Description

Adds a SoA version of VerletClustersColoringTraversal that uses one SoA per grid, so not one SoA per cluster as the other traversal, because a grid is essentially a particle cell which can be easily used to load and extract the particles. 
To be able to let the clusters interact with each other, the signatures of the SoAFunctor() that take one or two SoAs were changed to let the user determine which particles of each SoA should be considered. The implementation changes were minimal. The old signatures are still there, calling the new methods with the appropriate parameters. 

Another possibility for this change would be default parameters for the SoAFunctor() methods, but this is discouraged on virtual methods.

Furthermore, to get it compiling it was necessary to override all the old methods in the subclasses and just call the super implementation. I am not sure why, I just did it as it worked.
- [x] Look at why this was necessary and if it can be avoided (not necessary anymore since `SoAView`s are used)

**TL;DR**
  - Adds SoA to VerletClustersColoringTraversal
  - Changes (adds) Functor::SoAFunctor() signatures

## Related Pull Requests

- #295 Branched from here to separate the functor changes for SoA from the rest, so the rest can be merged sooner.
- #299 Could replace the functor refactoring done here. Then #299 needs to be merged first!

## Resolved Issues
 - Closes #311 

# How Has This Been Tested?

- Unit Tests

### CODE REVIEW

File: src/autopas/containers/verletClusterLists/traversals/VerletClustersColoringTraversal.h [hunk 1]
```diff
@@ -166,4 +214,19 @@ void VerletClustersColoringTraversal<ParticleCell, PairwiseFunctor, dataLayout,
   }
 }
 
+template <class ParticleCell, class PairwiseFunctor, DataLayoutOption dataLayout, bool useNewton3>
+void VerletClustersColoringTraversal<ParticleCell, PairwiseFunctor, dataLayout, useNewton3>::traverseClusterPairSoA(
+    Particle *clusterStart, Particle *neighborClusterStart) {
+  auto clusterView = _clusterToSoAViewMap[clusterStart];
+
+  // std::cout << autopas_get_thread_num() << " " << clusterStart << " " << neighborClusterStart << std::endl;
```
Comment (FG-TUM) [inline src/autopas/containers/verletClusterLists/traversals/VerletClustersColoringTraversal.h]: remove

Comment (HobbyProgrammer) [reply src/autopas/containers/verletClusterLists/traversals/VerletClustersColoringTraversal.h]: thx

File: src/autopas/containers/verletClusterLists/traversals/VerletClustersColoringTraversal.h [hunk 2]
```diff
@@ -30,6 +28,8 @@ class VerletClustersColoringTraversal : public CBasedTraversal<ParticleCell, Pai
                                         public VerletClustersTraversalInterface<typename ParticleCell::ParticleType> {
  private:
   using Particle = typename ParticleCell::ParticleType;
+  typedef typename VerletClusterMaths::index_t index_t;
```
[the remark this answers is not in the document]
Comment (HobbyProgrammer) [reply src/autopas/containers/verletClusterLists/traversals/VerletClustersColoringTraversal.h]: I had this like you suggested, but the compiler seems to do something strange here, as he gives the following warning:
`warning: declaration 'using index_t = std::size_t' does not declare anything`

I would just leave it like it is with typedef and without the warning, since the index_t type is removed in my refactoring in #319, anyway. (I hope that that is okay, by the way, but I didn't see the point, especially not when using auto.)

[the remark this answers is not in the document]
Comment (FG-TUM) [reply src/autopas/containers/verletClusterLists/traversals/VerletClustersColoringTraversal.h]: when it is removed anyways it is ok.
